gtda/mapper/tests/test_cover.py

```diff
@@ -94,9 +94,8 @@ def test_equal_interval_length(filter_values, n_intervals, overlap_frac):
         )
 
     # rounding precision
-    decimals = 10
-    assert len(set(np.floor((upper_limits - lower_limits) *
-                            decimals).tolist())) == 1
+    lengths = upper_limits - lower_limits
```

Comment:
I forgot what this was about, could you remind me?

Reply:
Yes! It's basically checking if all intervals in the cover have the same length. The previous version was failing due to numerical issues: we were rounding up `1-\epsilon` to 0 and `1+epsilon` to 1.

I changed the comparison to a more tolerant one: compute the range of lengths and see if that range is small.
Is that ok?

Reply:
God it, thanks!  Sounds good!  So the problem was with using floor!

Reply:
Worse. In general, the length can be any positive `float`, so regardless of what "threshold" you choose (0. for `np.floor`, or 0.5 for `np.round`), you might have rounding problems. I think it's simpler to check that the range of values of `length`, the interval `[min(lengths), max(lengths)]`, is very small (they are all similar).

Reply:
Great, thanks!